src/mqtt_client/mqtt_client.rs

```diff
@@ -112,73 +168,92 @@ where
 
         Ok(Self {
             mqtt,
-            clock,
+            state: sm::StateMachine::new(sm::Context::new(clock)),
             settings,
             settings_prefix,
             prefix: String::from(prefix),
-            subscribed: false,
-            iteration_state: None,
-            rx_timeout: None,
         })
     }
 
     fn handle_republish(&mut self) {
-        if let Some(timeout) = &self.rx_timeout {
-            if self.clock.try_now().unwrap() > *timeout {
-                log::info!("Republish duration lapsed. Publishing settings");
-                self.iteration_state.replace([0; MAX_RECURSION_DEPTH]);
-            }
+        if !self.mqtt.client.can_publish(QoS::AtMostOnce) {
+            return;
         }
 
-        let mut iteration_exhausted = false;
-        if let Some(ref mut iteration_state) = &mut self.iteration_state {
-            self.rx_timeout.take();
+        for topic in self
+            .settings
+            .into_iter::<{ MAX_TOPIC_LENGTH }>(&mut self.state.context_mut().republish_state)
+            .unwrap()
+        {
+            let mut data = [0; MESSAGE_SIZE];
+
+            // Note(unwrap): We know this topic exists already because we just got it from the
+            // iterator.
+            let len = self.settings.get(&topic, &mut data).unwrap();
 
+            let mut prefixed_topic: String<{ 2 * MAX_TOPIC_LENGTH + 1 }> = String::new();
```

Comment:
Increase it to 128 max topic length. Since we are juggling one topic at a time in all cases, the stack usage will be flat.

```suggestion
            let mut prefixed_topic: String<{ MAX_TOPIC_LENGTH }> = String::new();
```